contractcourt/htlc_outgoing_contest_resolver.go

```diff
@@ -198,6 +199,51 @@ func (h *htlcOutgoingContestResolver) Encode(w io.Writer) error {
 	return h.htlcTimeoutResolver.Encode(w)
 }
 
+// handOffHtlcTimeout returns the inner htlcTimeoutResolver so that we can
+// timeout outgoing htlc on chain. If the htlc is on our commitment and we need
+// to claim the htlc in two stages, we record stage one of the htlc claim here.
+func (h *htlcOutgoingContestResolver) handOffHtlcTimeout() (ContractResolver,
+	error) {
+
+	// If we do not have a signed timeout transaction, we do not need to
+	// record the first stage of the two step htlc claim process, because
+	// this htlc will simply be timed out directly from the remote party's
+	// commitment.
+	if h.htlcResolution.SignedTimeoutTx == nil {
+		return &h.htlcTimeoutResolver, nil
+	}
+
+	// If the SignedTimeoutTx is not nil, we are timing out the htlc in two
+	// stages from our own commitment. Here, we record the intermediate
+	// SignedTimeoutTx stage so that we have a complete record of all on
+	// chain transactions that were required to timeout this htlc.
+	spendTx := h.htlcResolution.SignedTimeoutTx
+	spendTxID := spendTx.TxHash()
+
+	// To get the amount of the sole input to our signed timeout tx, we
+	// lookup the input on our channel's commitment to get its value. We can
+	// look this outpoint up because the timeout tx spends from our commit.
+	outpoint := spendTx.TxIn[0].PreviousOutPoint
+	output, err := h.FindOwnOutput(&outpoint)
```

Comment:
Similar comment here re obtaining the value from the struct as is. Also applies to all other instances of this in the diff.